glide/src/main/java/com/google/accompanist/glide/Glide.kt

```diff
@@ -198,9 +205,11 @@ internal class GlideLoader(
             .addListener(listener)
             .into(target)
 
-        // If we're cancelled, clear the request from Glide
-        cont.invokeOnCancellation {
-            requestManager.clear(target)
+        // When we're cancelled/closed, clear the request from Glide
+        awaitClose {
+            // Unfortunately we need to post the clear. Glide will crash if the close happens
+            // in the same call frame as the Target calls above (happens in testing)
+            handler.post { requestManager.clear(target) }
```

Comment:
How about using coroutines for this? `private val handler = Handler(Looper.getMainLooper())` is a hardcoded dependency in this class, if you want to keep it like that, you could create a scope and use `Dispatchers.Main`

Reply:
Turns out that I don't actually need to post now. Removed. 🎉